### PR TITLE
Feature - add cacert for curl plugin

### PR DESCRIPTION
**Pull request by AndriiZ**
Currently it is not possible to set cacert for curl plugin, thus this setting may be hard to find. 

In my case, neither ~/.curlrc nor /etc/curlrc does not affect MPD curl plugin and https streams fail.
With this option, I have set cacert option in mpd.conf and https stream plays properly

### CODE REVIEW

File: src/input/plugins/CurlInputPlugin.cxx [hunk 1]
```diff
@@ -309,7 +311,7 @@ CurlInputStream::OnData(ConstBuffer<void> data)
 
 	if (data.size > GetBufferSpace()) {
 		AsyncInputStream::Pause();
-		throw CurlResponseHandler::Pause{};
+		throw CurlRequest::Pause();
```
Comment (MaxKellermann) [inline src/input/plugins/CurlInputPlugin.cxx]: Why this change?

File: src/input/plugins/CurlInputPlugin.cxx [hunk 2]
```diff
@@ -378,6 +380,7 @@ input_curl_init(EventLoop &event_loop, const ConfigBlock &block)
 
 	verify_peer = block.GetBlockValue("verify_peer", default_verify);
 	verify_host = block.GetBlockValue("verify_host", default_verify);
+        cacert = block.GetBlockValue("cacert");
```
Comment (MaxKellermann) [inline src/input/plugins/CurlInputPlugin.cxx]: Bad indent

File: src/input/plugins/CurlInputPlugin.cxx [hunk 3]
```diff
@@ -431,9 +434,11 @@ CurlInputStream::InitEasy()
 		request->SetOption(CURLOPT_PROXYUSERPWD,
 				   StringFormat<1024>("%s:%s", proxy_user,
 						      proxy_password).c_str());
+        if (cacert != nullptr)
+            request->SetOption(CURLOPT_CAINFO, cacert);
```
Comment (MaxKellermann) [inline src/input/plugins/CurlInputPlugin.cxx:438]: So CURL calls this CAINFO and you call it cacert... and it's undocumented!

File: src/input/plugins/CurlInputPlugin.cxx [hunk 4]
```diff
@@ -431,9 +434,11 @@ CurlInputStream::InitEasy()
...
 
-	request->SetVerifyPeer(verify_peer);
-	request->SetVerifyHost(verify_host);
+	request->SetOption(CURLOPT_SSL_VERIFYPEER, verify_peer ? 1L : 0L);
+	request->SetOption(CURLOPT_SSL_VERIFYHOST, verify_host ? 2L : 0L);
```
Comment (MaxKellermann) [inline src/input/plugins/CurlInputPlugin.cxx]: Why this change?

**Review by MaxKellermann — CHANGES_REQUESTED** [review]

File: src/input/plugins/CurlInputPlugin.cxx (hunk 4 above)
Comment (AndriiZ) [reply src/input/plugins/CurlInputPlugin.cxx]: Sorry, started with 0.22.4, swithed to main, so there is this change in the last commit

File: src/input/plugins/CurlInputPlugin.cxx (hunk 3 above)
Comment (AndriiZ) [reply src/input/plugins/CurlInputPlugin.cxx:438]: https://curl.se/libcurl/c/CURLOPT_CAINFO.html
CURLOPT_CAINFO - path to Certificate Authority (CA) bundle 
and from example 
curl_easy_setopt(curl, CURLOPT_CAINFO, "/etc/certs/cabundle.pem");

Do you think I should use another name? I have selected cacert because .curlrc has option cacert for CA bundle

File: src/input/plugins/CurlInputPlugin.cxx (hunk 1 above)
Comment (AndriiZ) [reply src/input/plugins/CurlInputPlugin.cxx]: Sorry, started with 0.22.4, switched to main, so there is this change in the last commit

File: src/input/plugins/CurlInputPlugin.cxx (hunk 2 above)
Comment (AndriiZ) [reply src/input/plugins/CurlInputPlugin.cxx]: Fixed

**Comment by MaxKellermann** [comment]
If you fix mistakes in a commit that is not yet merged, amend the commit instead of adding fixup commits. I can't review a PR with bad commits plus fixup commits.

**Comment by AndriiZ** [comment]
I'll create new one
